### ISSUE TITLE
Spans empty in `proc-macro2 >= 1.0.57`

### ISSUE THREAD

**Issue by MingweiSamuel** [hydro-project/hydroflow#729]
Currently working with `syn 2.0.14`, seems broken in `syn 2.0.18`

**Comment by MingweiSamuel** [comment]
broken by
```
 Updating proc-macro2 v1.0.56 -> v1.0.60
 Updating quote v1.0.26 -> v1.0.28
 Updating syn v2.0.16 -> v2.0.17
 ```

**Comment by MingweiSamuel** [comment]
broken by `proc-macro2 >= 1.0.57` 

https://github.com/dtolnay/proc-macro2/pull/383

**Comment by MingweiSamuel** [comment]
waiting for https://github.com/rust-lang/rust/pull/111571 to trickle through the ecosystem

**Comment by MingweiSamuel** [comment]
https://github.com/hydro-project/hydroflow/pull/801

probably breaks

https://github.com/hydro-project/hydroflow/blob/7dbd5e24d6e71cf8fab7c3ce09d5937c0f301456/website_playground/src/lib.rs#L70-L73

**Comment by MingweiSamuel** [comment]
Will check if upstream has changed, seems promising

**Comment by MingweiSamuel** [comment]
Not fixed as of 1.0.79
https://github.com/dtolnay/proc-macro2/pull/397
